### PR TITLE
[flutter_tools] migrate .packages to package_config

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

The package_config.json file has replaced the .packages file. While the package loader API we use can load either, we will need to eventually only support the JSON version.

Removes the global packages path and replaces it with the path in BuildInfo. Then updates many tests to prefer the package_config.json

### CODE REVIEW

**Review by zanderso — COMMENTED** [review]
Sorry, I'm still a bit unclear about what this PR does. Is `.packages` no longer supported after this PR? Is it supported with a flag? If so, who is still using it and where do we need to pass that flag, etc.?

**Comment by jonahwilliams** [comment]
Most of the dart tools have a sort of autodiscovery mechansm built in right now, where if you tell them `--packages=.packages`, they will first check for `.dart_tool/package_config.json` and only fall back to .packages. However, if they select .packages then you will get zero support for null safety or language versioning.

Users do not ever need to pass a --packages flag to us, but it they do we will still check for .packages. 

> What is the point of removing it then?

Not removing anything, just changing the default, and making our tests line up with what the tool is actually doing. This will have to change eventually because .packages will be deprecated and removed at some point

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart [hunk 1]
```diff
@@ -2701,7 +2701,7 @@ plugin1=${plugin1.path}
       legacySettingsDotGradleFiles.readAsStringSync().split(';EOF').map<String>((String body) => body.trim()),
       contains(templateSettingsDotGradle.readAsStringSync().trim()),
     );
-  });
+  }, skip: true); // TODO(jonahwilliams): This is an integration test and should be moved to the integration shard.
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/android/gradle_test.dart:2704]: This TODO probably needs a github issue if we don't already have one.

**Review by zanderso — APPROVED** [review]
lgtm

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/android/gradle_test.dart:2704]: https://github.com/flutter/flutter/issues/69975

**Comment by jonahwilliams** [comment]
Can't figure out frob issues, recreating

**Pull request closed without merging**